### PR TITLE
Closes #11. Closes #12

### LINKED ISSUE #11: Reverting an unsaved reminder
When actively editing a reminder: If the reminder has unsaved changes, the user will see a "Revert" button. When clicked it will automatically rollback the attributes of the reminder to its original, clean state.

**Pro tip**: I you're doing something that seems hard, maybe you should check the Ember Data API docs?

### LINKED ISSUE #12: Visual cues on unsaved reminders
When a reminder has changes that have not been saved to the database/server, the user should see some kind of visual indication in the sidebar of the application.

### PR DESCRIPTION
**Pull request by blakeworsley**
@stevekinney @martensonbj @brittanystoroz 

## Purpose

To show a revert button that appears when a reminder has been edited. Toggles a warning in the general reminder list to notify the users that a reminder is being edited.

## Approach

Revert button uses hasDirtyAttributes to either render or hide itself on the page. A warning is displayed in the reminder-list if the specific reminder hasDirtyAttributes.

### Learning

We once again learned that there was a very easy way to do what we wanted with Ember. We searched for hasDirtyAttributes in the ember docs and implemented it fairly easily with an action.

### Test coverage 

Wrote a test to determine if the revert button was appearing and disappearing when a reminder was being edited. Also tested that the reminder in the reminder-list was displaying a warning icon.

### CODE REVIEW

File: app/controllers/reminder-list/edit.js [hunk 1]
```diff
@@ -2,10 +2,17 @@ import Ember from 'ember';
 
 export default Ember.Controller.extend({
     actions: {
+
       updateReminder(id) {
         this.get('store').findRecord('reminder', id).then((reminder) => {
-          reminder.save();
+          reminder.save().then((reminder) => { this.transitionToRoute('reminder-list.reminder-detail', reminder.id); });
```
Comment (martensonbj) [inline app/controllers/reminder-list/edit.js]: You can just pass this method `reminder` not `reminder.id` - saves a trip to the database bc it defaults to caring about the `id`.

File: app/styles/app.scss [hunk 2]
```diff
@@ -139,20 +144,24 @@ button:hover {
...
   border: none;
-  margin: 0px;
+  margin: 0;
 }
 
 .reminder-item-date {
   font-family: $sans;
-  margin: 0px
+  margin: 0;
 }
```
Comment (martensonbj) [inline app/styles/app.scss]: Just a side-comment, would all of these "margin: 0" style specs have been covered by a reset.css file? Just a thought to clean up the css you actually care about.

File: app/templates/reminder-list.hbs [hunk 3]
```diff
@@ -2,8 +2,9 @@
   <ul class="reminder-list">
       {{#each model as |reminder|}}
         {{#link-to "reminder-list.reminder-detail" reminder tagName="li" class="reminder-list-item" }}
-            <h2 class="spec-reminder-item">{{reminder.title}}</h2>
-            <p class="reminder-item-date">{{reminder.date}}</p>
+          <span title="This reminder has unsaved changes!" class="dirty-warning-icon" hidden={{ unless reminder.hasDirtyAttributes 'true' }}>⚠️</span>
```
Comment (martensonbj) [inline app/templates/reminder-list.hbs]: Digging the tool-tip style notification! 👍

File: app/templates/reminder-list/edit.hbs [hunk 4]
```diff
@@ -4,8 +4,11 @@
     {{ input value=reminder.title class="reminder-title-input" placeholder="Insert Title" }} <br>
     {{ input value=reminder.date type="datetime-local" class="reminder-date-input" placeholder="Insert Date" }} <br>
     {{ textarea value=reminder.notes class="reminder-notes-input" placeholder="Insert Note" }} <br>
-    {{#link-to "reminder-list.reminder-detail" reminder}}
     <button class="reminder-submit-button" {{action "updateReminder" reminder.id on="click"}}>Submit</button>
-    {{/link-to}}
+    <button class="reminder-revert-button"
+      {{action "revertReminder" reminder.id on="click"}}
```
Comment (martensonbj) [inline app/templates/reminder-list/edit.hbs]: You can just send the `reminder` without the `reminder.id` here as well.

File: app/styles/app.scss (hunk 2 above)
Comment (Casey1449) [inline app/styles/app.scss]: I think so. There have been several places where a reset file would've saved some code...but probably added more code in places where we're relying on the default styling. I'm kind of scared to add one in now and then address all the unexpected consequences...is that an okay decision?

Comment (Casey1449) [inline app/styles/app.scss]: (ps, we didn't take all those 'px' suffixes by hand...our css beautifier did that.)

**Comment by Casey1449** [comment]
@martensonbj @stevekinney @brittanystoroz 

all comments addressed.

**Comment by martensonbj** [comment]
An additional general comment - try to keep an eye on consistency of spaces between curly braces. I'm seeing a combo of no space vs space and will address that in further pull requests.  @Casey1449 @blakeworsley

**Pull request merged**